src/OpenAL/OpenToolkit.OpenAL/Extensions/Creative.EnumerateAll/EnumerateAll.cs

```diff
@@ -0,0 +1,95 @@
+//
+// EnumerateAll.cs
+//
+// Copyright (C) 2019 OpenTK
+//
+// This software may be modified and distributed under the terms
+// of the MIT license. See the LICENSE file for details.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using OpenToolkit.Core.Extensions;
+using OpenToolkit.Core.Loader;
+
+namespace OpenToolkit.Audio.OpenAL.Extensions.Creative.EnumerateAll
+{
+    /// <summary>
+    /// Exposes the API in the EnumerateAll extension.
+    /// </summary>
+    [Api(ALC.Lib, typeof(OpenALLibraryNameContainer))]
+    public class EnumerateAll : ApiContainer<EnumerateAll>
+    {
+        private EnumerateAll()
+        {
+        }
+
+        /// <summary>
+        /// Checks whether the extension is present.
+        /// </summary>
+        /// <returns>Whether the extension was present or not.</returns>
+        public static unsafe bool IsExtensionPresent()
+        {
+            return ALC.IsExtensionPresent(IntPtr.Zero, "ALC_ENUMERATE_ALL_EXT");
+        }
+
+        /// <summary>
+        /// Checks whether the extension is present.
+        /// </summary>
+        /// <param name="device">The device to be queried.</param>
+        /// <returns>Whether the extension was present or not.</returns>
+        public static unsafe bool IsExtensionPresent(Device* device)
+        {
+            return ALC.IsExtensionPresent((IntPtr)device, "ALC_ENUMERATE_ALL_EXT");
```

Comment:
Hmm id like some implicit type conversion here, rather than casts everywhere, but I dont know how you guys feel about it.

Reply:
Yeah I also think that all of this `Device*` stuff should be removed. I can create a similar thing to `ALContextHandle` that just wraps an `IntPtr`.